src/ControlSystem/TimescaleTuner.hpp

```diff
@@ -0,0 +1,42 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cmath>
+
+#include "DataStructures/DataVector.hpp"
+#include "ErrorHandling/Error.hpp"
+
+/// \ingroup ControlSystemGroup
+/// Manages control system timescales
+class TimescaleTuner {
+ public:
+  TimescaleTuner(DataVector initial_timescale, double max_timescale,
+                 double min_timescale, double max_threshold,
+                 double min_threshold, double increase_factor,
+                 double decrease_factor) noexcept;
```

Comment:
We need some documentation of what all these parameters mean, either on the class or the constructor.

Reply:
I agree. I think most names are reasonably clear to read, aside from the thresholds.